src/Service/FileWatcher/Service_FileWatcher.rei

```diff
@@ -0,0 +1,8 @@
+[@deriving show]
+type event = {
+  path: string,
+  hasRenamed: bool,
+  hasChanged: bool,
+};
+
+let watch: (~path: string, ~onEvent: event => 'msg) => Isolinear.Sub.t('msg);
```

Comment:
This is the ideal subscription API IMO. Instead of offering a subscription with a hard-coded `msg` type that has to be mapped by the consumer to their own `msg` type, this take a set of mapping functions and does the mapping internally. Unfortunately this still requires a kludgy internal `msg` type, and fixing it requires using a record instead of fa first-class module I think. But this awkwardness only affects providers, so no rush.

Reply:
Makes sense, I like that better too - simpler for the consumer to use, versus needing the `Isolinear.Sub.map` everywhere.